src/server.js

```diff
@@ -57,6 +57,17 @@ export const startServer = async (listener) => {
     server.logger.info('Server started successfully')
     server.logger.info(`Access mock on http://localhost:${config.get('port')}`)
 
+    server.ext('onRequest', (request, h) => {
+      // log the domain of the email that was passed in the `email` header
+      if (request.headers.email) {
+        request.logger = request.logger.child({
+          tenant: { id: request.headers.email.split('@')[1] }
```

Comment:
You might need a unit test for this - what happens if there's no `email` header!?
```suggestion
          tenant: { id: request?.headers?.email?.split('@')[1] }
```

Reply:
I think the optional chaining should be